projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -194,11 +198,51 @@ export class ExplorerComponent {
     }
   }
 
+  private convertToFilterAttributes(attrArray: AttributeMetadata[]): FilterAttribute[] {
+    return attrArray.map(({ name, displayName, units, type, onlySupportsAggregation, onlySupportsGrouping }) => {
+      const applicableType = toFilterAttributeType(type);
+
+      return {
+        name: name,
+        displayName: displayName,
+        units: units,
+        type: applicableType,
+        onlySupportsAggregation: onlySupportsAggregation,
+        onlySupportsGrouping: onlySupportsGrouping
+      };
+    });
+  }
+
   public onContextUpdated(contextWrapper: ExplorerContextScope): void {
     this.attributes$ = this.metadataService.getFilterAttributes(contextWrapper.dashboardContext);
+    const listener = this.attributes$.subscribe(attributes => {
+      const lastTab = this.contextChangeSubject.getValue();
+      const newFilters = this.filters.map(eachFilter => {
+        // If the given filter has a different name for the selected tab, update the filter value
+        if (eachFilter.field in contextMapObject[lastTab]) {
+          const newFilter = this.filterChipService.autocompleteFilters(
```

Comment:
We should be able to do this without the chip service, that's an impl detail from the filter bar. Can we use the filter builder service (I hate how complex we made this!) instead just read the details out of the old one and write them into the new one. May need to lookup the new attribute, too - but we shouldn't have to create it, it's already one of the ones in the array coming in.